### PR TITLE
Support filtering referrals by status

### PR DESCRIPTION
**Pull request by gigxz**
## Description

Depends on hmis-warehouse PR: https://github.com/greenriver/hmis-warehouse/pull/4464
Closes https://github.com/open-path/Green-River/issues/6173 and https://github.com/open-path/Green-River/issues/5826

Adds a new functionality to TableFilters to let you specify the picklist options, in case you want to override the picklist that would be generated from the HmisEnum.

## Type of change
- [x] New feature (adds functionality)

## Checklist before requesting review
- [x] I have performed a self-review of my code
- [x] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [x] My code includes comments and/or descriptive variable names to help other engineers understand the intent (or not applicable)
- [x] My code follows the style guidelines of this project (eslint)
- [x] I have updated the documentation (or not applicable)
- [x] If it's not obvious how to test this change, I have provided testing instructions in this PR or the related issue

### CODE REVIEW

File: src/modules/admin/components/denials/AdminReferralDenials.tsx [hunk 1]
```diff
@@ -81,6 +94,15 @@ const AdminReferralDenials = () => {
...
           defaultPageSize={25}
+          filters={{
+            status: getReferralFilter([
+              ReferralPostingStatus.DeniedPendingStatus,
+              ReferralPostingStatus.DeniedStatus,
+            ]),
+          }}
+          defaultFilterValues={{
+            status: [ReferralPostingStatus.DeniedPendingStatus],
+          }}
```
Comment (gigxz) [inline src/modules/admin/components/denials/AdminReferralDenials.tsx:102]: Screenshot 2024-06-17 at 3 56 07 PM

File: src/modules/projects/components/tables/ProjectOutgoingReferralPostingsTable.tsx [hunk 2]
```diff
@@ -107,6 +109,17 @@ const ProjectOutgoingReferralPostingsTable: React.FC<Props> = ({
...
+        status: getReferralFilter([
+          ReferralPostingStatus.AssignedStatus,
+          ReferralPostingStatus.AcceptedPendingStatus,
+          ReferralPostingStatus.DeniedPendingStatus,
+          ReferralPostingStatus.AcceptedStatus,
+          ReferralPostingStatus.DeniedStatus,
+        ]),
+      }}
+      defaultPageSize={15}
+      paginationItemName='outgoing referral'
```
Comment (gigxz) [inline src/modules/projects/components/tables/ProjectOutgoingReferralPostingsTable.tsx:122]: Screenshot 2024-06-17 at 3 56 56 PM

File: src/modules/projects/components/tables/ProjectReferralPostingsTable.tsx [hunk 3]
```diff
@@ -95,7 +97,21 @@ const ProjectReferralPostingsTable: React.FC<Props> = ({
...
+          ReferralPostingStatus.AcceptedStatus,
+        ]),
+      }}
+      defaultFilterValues={{
+        status: [
+          ReferralPostingStatus.AssignedStatus,
+          ReferralPostingStatus.AcceptedPendingStatus,
+        ],
+      }}
+      paginationItemName='incoming referral'
```
Comment (gigxz) [inline src/modules/projects/components/tables/ProjectReferralPostingsTable.tsx:114]: Screenshot 2024-06-17 at 3 57 08 PM

File: src/modules/referrals/referralUtil.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+      return 4;
+    case ReferralPostingStatus.DeniedStatus:
+      return 5;
+    default:
+      return 10;
+  }
+};
+
+// Define a filter for ReferralPostingStatuses
+export const getReferralFilter = (
```
Comment (ttoomey) [inline src/modules/referrals/referralUtil.ts:23]: This is fine. Maybe could check how often the caller re-renders, maybe triggering table render. If so perhaps should memoize, something like:

```ts
export const useGetReferralFilter = (statuses) => {
  return useMemo( ()=> getReferralFilter(statuses), [statuses] )
}
```

**Review by ttoomey — APPROVED** [review]
looks good!

**Pull request merged**